File: courseSchedule_subjects_and_exercises/week_1_C/exercises/src/2mario_more.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_double_pyramid(int height);

int main(void)
{
  // Get height of pyramid
  int height = get_height();

  // Print double pyramid
  print_double_pyramid(height);
}

// Prompt user for height between 1 and 8
int get_height(void)
{
  int height;
  do
  {
    height = get_int("Height: ");
  } while (height < 1 || height > 8);
  return height;
}

// Print double pyramid with gap
void print_double_pyramid(int height)
{
  for (int row = 0; row < height; row++)
  {
    // Left pyramid spaces
    for (int space = 0; space < height - row - 1; space++)
    {
      printf(" ");
    }

    // Left pyramid hashes
    for (int hash = 0; hash < row + 1; hash++)
    {
      printf("#");
    }

    // Gap (2 spaces)
    printf("  ");

    // Right pyramid hashes
    for (int hash = 0; hash < row + 1; hash++)
    {
      printf("#");
    }

    // New line
    printf("\n");
  }
}

File: courseSchedule_subjects_and_exercises/week_6_0_python/exercises/src/2mario_more.py
from cs50 import get_int


def main():
    # Get pyramid height from user
    height = get_height()

    # Print the double pyramid
    print_double_pyramid(height)


def get_height():
    """Prompts user for height between 1 and 8, inclusive"""
    while True:
        try:
            height = get_int("Height: ")
            if 1 <= height <= 8:
                return height
            else:
                print("Height must be between 1 and 8, inclusive.")
        except ValueError:
            print("Please enter a valid integer.")


def print_double_pyramid(height):
    """Prints two half pyramids with specified height and 2-space gap"""
    for row in range(height):
        # Print spaces for left pyramid alignment
        for space in range(height - row - 1):
            print(" ", end="")

        # Print left pyramid hashes
        for hash in range(row + 1):
            print("#", end="")

        # Print 2-space gap between pyramids
        print("  ", end="")

        # Print right pyramid hashes
        for hash in range(row + 1):
            print("#", end="")

        # Move to next line
        print()


if __name__ == "__main__":
    main()


# mario.py => native to Python (without the CS50 library):


# def main():
#     # Get pyramid height from user
#     height = get_height()

#     # Print the double pyramid
#     print_double_pyramid(height)


# def get_height():
#     """Prompts user for height between 1 and 8, inclusive"""
#     while True:
#         try:
#             height = int(input("Height: "))
#             if 1 <= height <= 8:
#                 return height
#             else:
#                 print("Height must be between 1 and 8, inclusive.")
#         except ValueError:
#             print("Please enter a valid integer.")


# def print_double_pyramid(height):
#     """Prints two half pyramids with specified height and 2-space gap"""
#     for row in range(height):
#         # Calculate components for this row
#         spaces = " " * (height - row - 1)
#         hashes = "#" * (row + 1)

#         # Print the complete row: spaces + left hashes + gap + right hashes
#         print(f"{spaces}{hashes}  {hashes}")


# if __name__ == "__main__":
#     main()
